Rich Text: Prevent line breaks when disableLineBreaks is true

Infinite-Null opened the pull request:
Fixes: #67387

## What?
Modify `RichText` paste handling to correctly handle line breaks when `disableLineBreaks` is set to true

## Why?
Currently, when pasting text into a RichText component with disableLineBreaks={true}, line breaks are being converted to <br> tags instead of being stripped out as expected. This breaks the intended functionality of preventing line breaks in the component.

## Testing Instructions
1. Insert a Site Tagline block. Inside this block
2. Enter text, including line breaks, into the text area or you can use the below text.
```
Hello
Test
hi
```
3. Copy it and paste it into the Site Tagline block.

## Screencast
### Before Patch
https://github.com/user-attachments/assets/2f63563c-9df2-4313-ae6f-d01f05f867e8
 
### After Patch
https://github.com/user-attachments/assets/abda6d26-1e12-4e07-b4c8-60769bbbcc03

Infinite-Null committed:
Fix: Prevent line breaks when disableLineBreaks is true

markhowellsmead commented:
Looks good to me.

ellatrix commented on packages/block-editor/src/components/rich-text/event-listeners/paste-handler.js:
```diff
@@ -37,10 +38,15 @@ export default ( props ) => ( element ) => {
+		// When line breaks are disabled, convert all line breaks to a single space.
+		plainText = disableLineBreaks
+			? plainText.replace( /\r?\n/g, ' ' ).trim()
+			: plainText;
```
What if you paste HTML? This doesn't look like a complete solution. Could you modify `valueToInsert` within `pasteInline` instead? We should strip out all the `\n` there. There's a rich text replace function I believe.

Infinite-Null replied:
Hi @ellatrix,
Thank you for the thoughtful review. I've implemented the solution by modifying `valueToInsert` within `pasteInline` using the rich-text `replace` function as you recommended.

Could you please review the changes

Infinite-Null committed:
Fix: Line break handling in RichText paste event

ellatrix commented:
Thank you, this looks better. Could you add an e2e test?

Infinite-Null commented:
> Thank you, this looks better. Could you add an e2e test?

Sure @ellatrix,
I will open a follow-up PR for the e2e test

Infinite-Null committed:
Merge branch 'trunk' into fix/prevent-br-tags-with-disable-line-breaks

Infinite-Null committed:
Test: Add e2e test for RichText line break stripping in Site Title block

Infinite-Null committed:
Test: Add e2e test for RichText line break preservation

Infinite-Null commented:
Hi @ellatrix,
I have completed the end-to-end test for this PR. Please have a look at it at your convenience. Thank You!

## Screencast:

https://github.com/user-attachments/assets/ecd7596d-78b8-4b0c-a583-c6f5114a33f2

Infinite-Null commented:
Hi @ellatrix,
Just wanted to bump this PR - still looking for your review when you have a moment. Thanks!

ellatrix commented on test/e2e/specs/editor/various/rich-text.spec.js:
```diff
@@ -8,6 +8,58 @@ test.describe( 'RichText (@firefox, @webkit)', () => {
+		const multiLineContent = 'First<br/>Second<br/>Third';
+		pageUtils.setClipboardData( {
+			plainText: multiLineContent,
+			html: multiLineContent.replace( /\n/g, '<br>' ),
```
`multiLineContent` is already HTML? Shouldn't it be the other way around?

Infinite-Null replied:
Hi @ellatrix, Thank you for the review. I have fixed this in my changes by setting `multiLineContent` as actual plain text with line breaks: `First\nSecond\nThird`, Also made the change in `should not strip line breaks when pasting with disableLineBreaks diabled` test.

Infinite-Null committed:
Test: Fixed clipboard data simulation for RichText line break tests

- Fixed the multiLineContent format to use actual line breaks instead of HTML tags
- Updated HTML conversion to consistently use <br/> tags

Infinite-Null commented:
Hi @ellatrix,
I have made the necessary changes please review it at your convenience.

Thank You!

Infinite-Null committed:
Merge branch 'trunk' into fix/prevent-br-tags-with-disable-line-breaks

Infinite-Null commented:
Hi @ellatrix,
Just wanted to bump this PR - still looking for your review when you have a moment. Thank You!

Infinite-Null commented:
Hi @ellatrix,
Just wanted to bump this PR - still looking for your review when you have a moment. Thank You!

Infinite-Null commented:
Hi @ellatrix,
Just wanted to bump this PR - still looking for your review when you have a moment. Thank You!

Infinite-Null commented:
Hi @ellatrix,
Just wanted to bump this PR - still looking for your review when you have a moment. Thank You!

Infinite-Null committed:
Merge branch 'trunk' into fix/prevent-br-tags-with-disable-line-breaks

Infinite-Null committed:
Merge branch 'trunk' into fix/prevent-br-tags-with-disable-line-breaks

Infinite-Null committed:
Merge branch 'trunk' into fix/prevent-br-tags-with-disable-line-breaks

ellatrix commented on test/e2e/specs/editor/various/rich-text.spec.js:
```diff
@@ -8,6 +8,58 @@ test.describe( 'RichText (@firefox, @webkit)', () => {
+		editor,
+		pageUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/site-title' } );
```
This could be a problem when we ever change site title to allow line breaks. A more stable approach would be to register and insert a minimal block client side containing a RichText element with the required attributes?

ellatrix reviewed (approved):
Looks good

Infinite-Null committed:
test: decouple rich text line break test from core blocks by registering a temporary custom block

Infinite-Null committed:
test: refactor formatting and update assertion in rich-text E2E test
